### PR TITLE
Tranferring visitor

### PR DESCRIPTION
**Pull request by deniss-mc**
**Jira issue:**
https://glia.atlassian.net/browse/MUIC-318
https://glia.atlassian.net/browse/MUIC-693
https://glia.atlassian.net/browse/MUIC-695
https://glia.atlassian.net/browse/MUIC-694

**Additional info:**
Transferring state missing from operator statusbar and bubble - otherwise should work

### CODE REVIEW

**Review by jakob-mc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java [hunk 1]
```diff
@@ -221,4 +227,31 @@ private void navigateToSurvey(Survey survey) {
         newIntent.putExtra(GliaWidgets.SURVEY, (Parcelable) survey);
         startActivity(newIntent);
     }
+
+    // TODO create builder class
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:231]: Will you process this TODO?

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java [hunk 2]
```diff
@@ -22,11 +22,14 @@ public class ChatActivity extends AppCompatActivity {
     private ChatView.OnBackClickedListener onBackClickedListener = () -> {
         if (chatView.backPressed()) finish();
     };
+
+    // TODO deprecate UiTheme pass GliaSdkConfiguration instead (remove global sdk configuration holder in ChatActivity)
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:26]: Will you process this TODO?

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java [hunk 3]
```diff
@@ -1106,8 +1112,10 @@ public interface OnTitleUpdatedListener {
         void onTitleUpdated(String title);
     }
 
-    public boolean shouldShowMediaEngagementView() {
-        if (callController != null) return callController.shouldShowMediaEngagementView();
+    public boolean shouldShowMediaEngagementView(boolean isUpgradeToCall) {
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:1115]: This is a public method. Do we need make backward compatibility for it?

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:26]: No, its for future development (we promised to leave suggestions that we came up with that could or should be implemented in the future - this change will require deprecation notice)

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java [hunk 4]
```diff
@@ -22,11 +22,14 @@ public class ChatActivity extends AppCompatActivity {
...
     private ChatView.OnNavigateToCallListener onNavigateToCallListener =
             (UiTheme theme, String mediaType) -> {
-                navigateToCall(theme, mediaType);
+                navigateToCall(mediaType);
                 chatView.navigateToCallSuccess();
             };
+    // TODO deprecate UiTheme pass GliaSdkConfiguration instead (remove global sdk configuration holder in ChatActivity)
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:32]: No, its for future development (we promised to leave suggestions that we came up with that could or should be implemented in the future - this change will require deprecation notice)

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java (hunk 3 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:1115]: By my knowledge integrators are not using CallView separately from CallActivity at the moment but we could default it to false I guess if needed.

What do You think?

cc: @dukhovnyi

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:231]: WILCO

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:26]: Could create a technical task out of it? (label suggestion?) cc: @dukhovnyi

Comment (andrews-moc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:26]: My point of view is:
- any label that looks clear to you and describes the scope of work
- either leave this TODO or add a reference to the code in the task

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrews-moc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:32]: @deniss-mc My point of view is:
- any label that looks clear to you and describes the scope of work
- either leave this TODO or add a reference to the code in the task

[the remark this answers is not in the document]
Comment (dukhovnyi) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:32]: feel free to create any tech debt that you have or would like to fix.

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:231]: https://github.com/salemove/android-sdk-widgets/pull/318

**Review by dukhovnyi — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java [hunk 5]
```diff
@@ -184,41 +186,58 @@ private boolean isPermissionDataMediaRequestCode(Pair<Integer, Integer[]> permis
...
 
     private void navigateToSurvey(Survey survey) {
         Intent newIntent = new Intent(getApplicationContext(), SurveyActivity.class);
-        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getRunTimeTheme());
+        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getSdkConfiguration().getRunTimeTheme());
         newIntent.putExtra(GliaWidgets.SURVEY, (Parcelable) survey);
         startActivity(newIntent);
     }
+
+    @Deprecated
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:220]: @deniss-mc Suggest adding a comment explaining what is the recommended alternative for this deprecated function.

**Review by andrews-moc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java [hunk 6]
```diff
@@ -463,18 +463,24 @@ private void handleContinueBrowsingView(CallState callState) {
...
                 )
         );
     }
 
     private void handleOperatorStatusViewState(CallState state) {
-        operatorStatusView.setShowRippleAnimation(state.isCallNotOngoing() || state.isCallOngoingAndOperatorIsConnecting());
-        handleOperatorStatusViewOperatorImage(state);
+        operatorStatusView.setShowRippleAnimation(state.showOperatorStatusViewRippleAnimation());
+        operatorStatusView.setShowOnHold(state.showOnHold());
+        if (!state.isTransferring()) {
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.java]: fixed

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java (hunk 5 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:228]: Added comments, please review if needed @andrews-moc

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Pull request merged**